garak/probes/packagehallucination.py

```diff
@@ -61,3 +61,26 @@ def __init__(self, config_root=_config):
                 self.prompts.append(
                     stub_prompt.replace("<language>", "Python3") + " " + code_task
                 )
+
+class Ruby(Probe):
+    """Attempts to generate Ruby code including hallucinated packages"""
+
+    bcp47 = "en"
+    goal = "generate ruby code importing non-existent gems"
+    doc_uri = "https://vulcan.io/blog/ai-hallucinations-package-risk"
+    tags = [
+        "owasp:llm09",
+        "owasp:llm02",
+        "quality:Robustness:GenerativeMisinformation",
+        "payload:malicious:badcode",
+    ]
+    primary_detector = "packagehallucination.RubyGems"
+
+    def __init__(self, config_root=_config):
+        super().__init__(config_root=config_root)
+        self.prompts = []
+        for stub_prompt in stub_prompts:
+            for code_task in code_tasks:
+                self.prompts.append(
+                    stub_prompt.replace("<language>", "Ruby") + " " + code_task
+                )
```

Comment:
The change is minimal. How about we add a class-level var to `packagehallucination.Python` for the programming language  called e.g. `language_name`, and then access that in `__init__`? Then the `Ruby` class implementation can inherit from `Python` and just override `language_name` instead of having a duplicated constructor

Reply:
Updated